### ISSUE TITLE
Show number of lines folded

### ISSUE THREAD

**Issue by uber6** [microsoft/vscode#130250]
Current when a block is folded it currently only show ellipses, which I think it could be improved by show the lines of code in the block.

e.g. (using Python as an example)

```python
def printthis(printme):
    print(printme)
    print("how did it print?")
```

When the code above is folded, it is represented as:

```python
def printthis(printme): ...
```

Instead of just "...", it would be very useful to display how many lines are folded.

```python
def printthis(printme): ...(2 lines)...
```

With short block it is not too difficult to identify the number lines folded by looking at the line number, but for a larger blocks, it would aid significantly by showing the number of line folded.

**Comment by uber6** [comment]
Thanks bot, but none of the recommendations are like mine.

**Comment by NathanielRN** [comment]
Also related: #124177

**Comment by NathanielRN** [comment]
> Instead of just "...", it would be very useful to display how many lines are folded.

Just a small suggestion now that we hit the votes 🙂 Instead of "number of lines", I think it would be great if it said "number of items".

For example:

```json
"CloudWatchMetrics": [
    {
        "Namespace": "test",
        "Dimensions": [
            [
                "process.pid",
                "process.executable.name",
                "process.executable.path",
                "process.command",
                "process.command_line"
            ]
        ],
        "Metrics": [
            {
                "Name": "process.memory.physical_usage",
                "Unit": "Bytes"
            },
            {
                "Name": "process.memory.virtual_usage",
                "Unit": "Bytes"
            }
        ]
    }
],
```

would become

```json
"CloudWatchMetrics": [
    { ...(1 item)...
    }
],
```

Especially since you can already see the # of lines in the sidebar:

![image](https://user-images.githubusercontent.com/23139455/129965453-ad343749-f881-4b92-b28b-237b6c987f5b.png)

**Comment by lifez** [comment]
@aeschli  Can I start on this issue?

**Comment by lapaz17** [comment]
@aeschli  I think this issue is better suited for an extension. What do you think?

**Comment by aeschli** [comment]
I don't think this can be done in an extension as there's no API to access all folded regions.
The change would be in the core. 
Check out https://github.com/microsoft/vscode/tree/main/src/vs/editor/contrib/folding 
The `inline-folded` is the css class that adds the `...`.

**Comment by iKlsR** [comment]
This would be very useful, after using it in the insomnia client can't imagine not having it before.
<img width="239" alt="test" src="https://user-images.githubusercontent.com/741363/137535010-b93d86e0-3a2f-4549-b927-7ede06e2173c.PNG">

**Comment by jzyrobert** [comment]
So I remembered I had given this a go a while ago, except it didn't work properly:
![image](https://user-images.githubusercontent.com/20613660/176317602-b92675c2-ba4b-45d5-a06d-ae147bd6bf77.png)

The decoration wouldn't update in real-time until you edited the document, and it would randomly freeze VS Code when you clicked fold/unfold too often. But if someone wants to pick it up, my WIP code changes are in this commit: https://github.com/microsoft/vscode/commit/67c4de2a3c533f2eeb1beaf4597a40c0722dc3f5

**Comment by ziming-g** [comment]
Giving this a try.

**Comment by ziming-g** [comment]
Hmm, @jzyrobert 's WIP commit doesn't seem to want to work (at least, on my end) when I apply the contents to a new fork from December of `main` branch. The decoration gets created just fine but `changeDecorationOptions` doesn't apply the `inline-folded-count` element (the thing that includes the injected text for the lines). 

Also, sometimes removing sections of content between brackets or close then reopening a file crashes the editor very badly... I'll look into these issues some more in a few days but anyone please feel free to work on this as well.

**Comment by mtbaqer** [comment]
Hey everyone, I managed to make an extension that solves this problem along with other many requested folding-related issues. You can find it here:

https://marketplace.visualstudio.com/items?itemName=MohammadBaqer.better-folding

A demo of the extension in action:

https://user-images.githubusercontent.com/32401657/214105273-d3a70773-34ff-448a-9e4b-697b0f13ade3.mov

**Comment by zorapeteri** [comment]
I've wanted to comment this on https://github.com/microsoft/vscode/issues/124177 but since it's closed and this issue is mention on it I'm sharing here: 

in case anyone needs this for viewing JSON files in vscode and wants to see how many keys there are in an object (like with codemirror/insomnia):
I've found an extension that albeit doesn't show the numbers in the editor itself, it does show you how many keys objects have and some other cool stuff, without having to leave vscode.

https://marketplace.visualstudio.com/items?itemName=ZainChen.json
